pkg/pool-utils/contracts/test/MaliciousQueryReverter.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+contract MaliciousQueryReverter {
+    bool public revertMaliciously;
+
+    function setRevertMaliciously(bool enabled) external {
+        revertMaliciously = enabled;
+    }
+
+    function spoofJoinExitQueryRevert() public pure {
+        uint256[] memory tokenAmounts = new uint256[](2);
+        tokenAmounts[0] = 1;
+        tokenAmounts[1] = 2;
+
+        uint256 bptAmount = 420;
```

Comment:
_the number_

Reply:
It's like leaving known bugs in a commit for audit, gotta know if the review is thorough. :smile: